feat/arch-user-repository

claymorwan opened the pull request:
<!-- Replace "Website" with a markdown link to the website that you have themed. -->

## 🎉 Theme for Arch user repository 🎉

<!--
You should give a short description of the website that you have themed.
E.g. YouTube is a video sharing platform that allows users to upload, view, and share videos.

You should also attach some screenshots of the themed website, show it off!
-->
The Arch user repository (aur) is a community-driven repository of packages for Arch Linux.
![image](https://github.com/user-attachments/assets/2eb7075f-e7a2-4655-8f08-eddd61adbad1)


## 💬 Additional Comments 💬

<!--
Include any difficulties you had theming this port, or any general comments that would be useful for the reviewer to know.
Feel free to leave this section empty if you don't have anything more to say.
-->
I had a bit of struggle making this theme since it was my first time working with userstyles, but I made it in a day so it's fine. My only problem at the moment is changing the logo for a themed one, or at least a transparent one.

## 🗒 Checklist 🗒

- [x] I have read and followed Catppuccin's [submission guidelines](https://github.com/catppuccin/userstyles/blob/main/docs/userstyle-creation.md).
- [x] I have made a new directory underneath `/styles/<name-of-website>` containing the contents of the [`/template`](https://github.com/catppuccin/userstyles/blob/main/template/) directory.
  - [x] I have ensured that the new directory is in **lower-kebab-case**.
  - [x] I have followed the template and kept the preprocessor as [LESS](https://lesscss.org/#overview).
- [x] I have made sure to update the
      [`userstyles.yml`](https://github.com/catppuccin/userstyles/blob/main/scripts/userstyles.yml)
      file with information about the new userstyle.
- [x] I have included the following files:
  - [x] `catppuccin.user.css` - all the CSS for the userstyle, based on the
        template.
  - [x] `preview.webp` - composite image of all four individual flavor screenshots (taken with the default accent color of mauve) stitched together, generated via [Catwalk](https://github.com/catppuccin/catwalk).

uncenter commented on scripts/userstyles.yml:
```diff
@@ -170,6 +173,13 @@ userstyles:
         > This theme applies to [AniChart](https://anichart.net/) as well, which is an extension of AniList.
     current-maintainers: [*anubisnekhet]
+  arch-user-repository:
+    name: Arch user repository
```
```suggestion
    name: Arch User Repository
```

uncenter commented on scripts/userstyles.yml:
```diff
@@ -170,6 +173,13 @@ userstyles:
     current-maintainers: [*anubisnekhet]
+  arch-user-repository:
+    name: Arch user repository
+    categories: [development, search_engine, library]
```
```suggestion
    categories: [development]
```

uncenter commented on scripts/userstyles.yml:
```diff
@@ -1009,4 +1019,5 @@ userstyles:
           answer: "It's available in case you have an OLED display. If you have one, you might want to enable this."
     current-maintainers: [*isabelroses, *uncenter]
     past-maintainers: [*elkrien]
+
```
```suggestion
```

uncenter commented on styles/arch-user-repository/catppuccin.user.css:
```diff
@@ -0,0 +1,321 @@
+            background-color: @surface0 !important
+        }
+        
+      /* If you need to specify an overwrite specific to a flavor
+      *  (e.g. only when the flavor is set to latte), you can use a `when` guard clause. */
+      & when (@lookup = latte) {
+        /* ... */
+      }
+      /* You can also negate the statement above by using
+      *  a `when not` guard clause. */
+      & when not (@lookup = latte) {
+        /* ... */
+      }
```
Please remove the unused template stuff.

```suggestion
```

uncenter commented on styles/arch-user-repository/catppuccin.user.css:
```diff
@@ -0,0 +1,321 @@
+  }
+  
+  // vim:ft=less
+  
```
Trailing newline please! See https://editorconfig.org/ to configure your editor to respect .editorconfig settings (like trailing newlines).

uncenter commented on styles/arch-user-repository/catppuccin.user.css:
```diff
@@ -0,0 +1,321 @@
+  
+  
+  
+  /* prettier-ignore */
+  @catppuccin: {
+    @latte:     { @rosewater: #dc8a78; @flamingo: #dd7878; @pink: #ea76cb; @mauve: #8839ef; @red: #d20f39; @maroon: #e64553; @peach: #fe640b; @yellow: #df8e1d; @green: #40a02b; @teal: #179299; @sky: #04a5e5; @sapphire: #209fb5; @blue: #1e66f5; @lavender: #7287fd; @text: #4c4f69; @subtext1: #5c5f77; @subtext0: #6c6f85; @overlay2: #7c7f93; @overlay1: #8c8fa1; @overlay0: #9ca0b0; @surface2: #acb0be; ...
+    @frappe:    { @rosewater: #f2d5cf; @flamingo: #eebebe; @pink: #f4b8e4; @mauve: #ca9ee6; @red: #e78284; @maroon: #ea999c; @peach: #ef9f76; @yellow: #e5c890; @green: #a6d189; @teal: #81c8be; @sky: #99d1db; @sapphire: #85c1dc; @blue: #8caaee; @lavender: #babbf1; @text: #c6d0f5; @subtext1: #b5bfe2; @subtext0: #a5adce; @overlay2: #949cbb; @overlay1: #838ba7; @overlay0: #737994; @surface2: #626880; ...
+    @macchiato: { @rosewater: #f4dbd6; @flamingo: #f0c6c6; @pink: #f5bde6; @mauve: #c6a0f6; @red: #ed8796; @maroon: #ee99a0; @peach: #f5a97f; @yellow: #eed49f; @green: #a6da95; @teal: #8bd5ca; @sky: #91d7e3; @sapphire: #7dc4e4; @blue: #8aadf4; @lavender: #b7bdf8; @text: #cad3f5; @subtext1: #b8c0e0; @subtext0: #a5adcb; @overlay2: #939ab7; @overlay1: #8087a2; @overlay0: #6e738d; @surface2: #5b6078; ...
+    @mocha:     { @rosewater: #f5e0dc; @flamingo: #f2cdcd; @pink: #f5c2e7; @mauve: #cba6f7; @red: #f38ba8; @maroon: #eba0ac; @peach: #fab387; @yellow: #f9e2af; @green: #a6e3a1; @teal: #94e2d5; @sky: #89dceb; @sapphire: #74c7ec; @blue: #89b4fa; @lavender: #b4befe; @text: #cdd6f4; @subtext1: #bac2de; @subtext0: #a6adc8; @overlay2: #9399b2; @overlay1: #7f849c; @overlay0: #6c7086; @surface2: #585b70; ...
+  }
+  
+  // vim:ft=less
```
```suggestion
/* prettier-ignore */
@catppuccin: {
    @latte:     { @rosewater: #dc8a78; @flamingo: #dd7878; @pink: #ea76cb; @mauve: #8839ef; @red: #d20f39; @maroon: #e64553; @peach: #fe640b; @yellow: #df8e1d; @green: #40a02b; @teal: #179299; @sky: #04a5e5; @sapphire: #209fb5; @blue: #1e66f5; @lavender: #7287fd; @text: #4c4f69; @subtext1: #5c5f77; @subtext0: #6c6f85; @overlay2: #7c7f93; @overlay1: #8c8fa1; @overlay0: #9ca0b0; @surface2: #acb0be; @surface1: #bcc0cc; @surface0: #ccd0da; @base: #eff1f5; @mantle: #e6e9ef; @crust: #dce0e8; };
    @frappe:    { @rosewater: #f2d5cf; @flamingo: #eebebe; @pink: #f4b8e4; @mauve: #ca9ee6; @red: #e78284; @maroon: #ea999c; @peach: #ef9f76; @yellow: #e5c890; @green: #a6d189; @teal: #81c8be; @sky: #99d1db; @sapphire: #85c1dc; @blue: #8caaee; @lavender: #babbf1; @text: #c6d0f5; @subtext1: #b5bfe2; @subtext0: #a5adce; @overlay2: #949cbb; @overlay1: #838ba7; @overlay0: #737994; @surface2: #626880; @surface1: #51576d; @surface0: #414559; @base: #303446; @mantle: #292c3c; @crust: #232634; };
    @macchiato: { @rosewater: #f4dbd6; @flamingo: #f0c6c6; @pink: #f5bde6; @mauve: #c6a0f6; @red: #ed8796; @maroon: #ee99a0; @peach: #f5a97f; @yellow: #eed49f; @green: #a6da95; @teal: #8bd5ca; @sky: #91d7e3; @sapphire: #7dc4e4; @blue: #8aadf4; @lavender: #b7bdf8; @text: #cad3f5; @subtext1: #b8c0e0; @subtext0: #a5adcb; @overlay2: #939ab7; @overlay1: #8087a2; @overlay0: #6e738d; @surface2: #5b6078; @surface1: #494d64; @surface0: #363a4f; @base: #24273a; @mantle: #1e2030; @crust: #181926; };
    @mocha:     { @rosewater: #f5e0dc; @flamingo: #f2cdcd; @pink: #f5c2e7; @mauve: #cba6f7; @red: #f38ba8; @maroon: #eba0ac; @peach: #fab387; @yellow: #f9e2af; @green: #a6e3a1; @teal: #94e2d5; @sky: #89dceb; @sapphire: #74c7ec; @blue: #89b4fa; @lavender: #b4befe; @text: #cdd6f4; @subtext1: #bac2de; @subtext0: #a6adc8; @overlay2: #9399b2; @overlay1: #7f849c; @overlay0: #6c7086; @surface2: #585b70; @surface1: #45475a; @surface0: #313244; @base: #1e1e2e; @mantle: #181825; @crust: #11111b; };
}
  
// vim:ft=less
```

GenShibe commented:
In terms of the logo, my first thought is to take the SVG off of the arch wiki style, and then using that as a base to add the words "user repository"

claymorwan commented:
Ok I think I got everything fixed

isabelroses commented on styles/arch-user-repository/catppuccin.user.css:
```diff
@@ -0,0 +1,310 @@
+@name Arch user repository Catppuccin
+@namespace github.com/catppuccin/userstyles/styles/arch-user-repository
+@homepageURL https://github.com/catppuccin/userstyles/tree/main/styles/arch-user-repository
+@version 0.9.0
```
we don't need to change this until it reaches main
```suggestion
@version 0.1.0
```

isabelroses commented on styles/arch-user-repository/catppuccin.user.css:
```diff
@@ -0,0 +1,310 @@
+==/UserStyle== */
+
+@-moz-document domain('aur.archlinux.org') {
+    /* This section applies the light/dark themes automatically based on the user's system theme.
+       Use the below example CSS to start if the website you are theming doesn't have a light/dark mode toggle.
+       For an actual example of this, see https://github.com/catppuccin/userstyles/blob/39288b593278e5efa48007d5fc001292cf910672/styles/brave-search/catppuccin.user.css#L16-L29.
```
```suggestion
```

isabelroses commented on styles/arch-user-repository/catppuccin.user.css:
```diff
@@ -0,0 +1,310 @@
+      }
+    }
+
+    /* If the website you are theming *does* have a light/dark mode toggle,
+       you can conditionally enable light/dark flavors by targeting whatever
+       attribute that website uses to indicate the theme. *The example below gives you an idea of what
+       to look for. The website you are theming may use a class-based approach (e.g. :root.theme-dark),
+       or any of the myriad of other methods like this.*
+       Use the below example CSS to start if this is the case.
+       For an actual example of this, see https://github.com/catppuccin/userstyles/blob/39288b593278e5efa48007d5fc001292cf910672/styles/youtube/catppuccin.user.css#L21-L30.
+    */
```
```suggestion
```

isabelroses reviewed (commented):
First off, thanks for the PR :).

It would be nice to theme visited links.

The red here is not catppuccin themed.
![image](https://github.com/user-attachments/assets/bb52681a-b52d-42c2-9484-371d356d8be3)

Package keywords are a bit hard to read, and would likely be worse if my accent matched the blue, could we use crust on those.
![image](https://github.com/user-attachments/assets/9ee13d8f-cb20-4538-81ff-b088f1c4892a)

claymorwan commented:
I think I fixed everything now, tho idk about that conflict, tried to fix it by adding what was conflicting to the file

claymorwan commented:
> I think I fixed everything now, tho idk about that conflict, tried something but didn't worked

claymorwan commented:
why's it broken again

uncenter commented:
> why's it broken again

If you look at the logs, or run the linter locally - `deno task lint` - it will tell you the error.

claymorwan commented:
ok NOW it should be working, I did exactly what deno told me to do

isabelroses reviewed (changes requested):
I'm noticing some borders that are not themed, this also goes for most horizontal rules too.
e.g.
![image](https://github.com/user-attachments/assets/558cf3dc-eed9-4a0a-9087-aeafda2a9257)

The accounts page currently has unthemed text making it hard to read, [reproduction url](https://aur.archlinux.org/account/isabel)
![image](https://github.com/user-attachments/assets/721fccfc-2914-4a87-b3b9-69b47b4b7a75)

I see theres some code, first check if we have a port for it, and then import it. A good example is chatgpt. [repro url](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=aur-auto-vote-git).

Also noticing some none synatx highlighted colors on code too. [repro url](https://aur.archlinux.org/cgit/aur.git/diff/PKGBUILD?h=aur-auto-vote-git)
![image](https://github.com/user-attachments/assets/b39f7cd9-c5ea-423d-8205-8eae6a48f402)

Theres also a stats page for packages with an unthemed re. [repro url](https://aur.archlinux.org/cgit/aur.git/stats/PKGBUILD?h=aur-auto-vote-git)
![image](https://github.com/user-attachments/assets/64c27741-140b-4b81-aaf4-b6aef9ff15ca)

claymorwan commented:
So like I try to theme borders, acc page and syntax highlighting right? At first I left red colors and stuff to the default cuz I didn't know if I could use the catppuccin colors (using red for someone who has red accents color maybe would look weird idk) but ig it's fine
Doing that when I'm home

isabelroses commented on styles/arch-user-repository/catppuccin.user.css:
```diff
@@ -0,0 +1,353 @@
+  // Links
+        a {
+            color: @accent-color !important;
+            text-decoration: none !important;
```
why are we doing this?

claymorwan replied:
idk but it works

uncenter replied:
We shouldn't change non-color things, like text-decoration. Why is the change needed?

isabelroses reviewed (changes requested):
Just noticed a little more green, [repro url](https://aur.archlinux.org/cgit/aur.git/log/?h=crystal-nightly)
![image](https://github.com/user-attachments/assets/b55474e3-e50f-45d0-8659-2f215aa4e407)

claymorwan commented:
what did I do this time

claymorwan commented:
Oki is it gud now

uncenter commented:
I wonder if we should instead create a unified theme for the various Arch Linux pages - we already have a theme for the wiki, we could literally copy the `@-moz-document` part all the way down and paste it in that one, rename it... other pages like the home page https://archlinux.org/, https://archlinux.org/packages/, https://bbs.archlinux.org/, https://security.archlinux.org/, could be themed as well.

claymorwan commented:
> I wonder if we should instead create a unified theme for the various Arch Linux pages - we already have a theme for the wiki, we could literally copy the `@-moz-document` part all the way down and paste it in that one, rename it... other pages like the home page https://archlinux.org/, https://archlinux.org/packages/, https://bbs.archlinux.org/, https://security.archlinux.org/, could be themed as well.

Well one thing is sure, the wiki theme doesn't work on the aur page, so that would mean making a big file which each theme applying to it's page. I don't think all these website have similar enough HTML and CSS, execpt for the navbar which is the same on every of them

uncenter commented:
> > I wonder if we should instead create a unified theme for the various Arch Linux pages - we already have a theme for the wiki, we could literally copy the `@-moz-document` part all the way down and paste it in that one, rename it... other pages like the home page [archlinux.org](https://archlinux.org/), [archlinux.org/packages](https://archlinux.org/packages/), [bbs.archlinux.org](https://bbs.archlinux.org/), [security.archlinux.org](https://security.archlinux.org/), could be themed as well.
> 
> Well one thing is sure, the wiki theme doesn't work on the aur page, so that would mean making a big file which each theme applying to it's page. I don't think all these website have similar enough HTML and CSS, execpt for the navbar which is the same on every of them

I mean that's kind of the point. I think we *should* theme those other pages too for consistency, and merge this AUR and the existing wiki one together as well.

claymorwan commented:
> > > I wonder if we should instead create a unified theme for the various Arch Linux pages - we already have a theme for the wiki, we could literally copy the `@-moz-document` part all the way down and paste it in that one, rename it... other pages like the home page [archlinux.org](https://archlinux.org/), [archlinux.org/packages](https://archlinux.org/packages/), [bbs.archlinux.org](https://bbs.archlinux.org/), [security.archlinux.org](https://security.archlinux.org/), could be themed as well.
> > 
> > 
> > Well one thing is sure, the wiki theme doesn't work on the aur page, so that would mean making a big file which each theme applying to it's page. I don't think all these website have similar enough HTML and CSS, execpt for the navbar which is the same on every of them
> 
> I mean that's kind of the point. I think we _should_ theme those other pages too for consistency, and merge this AUR and the existing wiki one together as well.

Then i can try on my end see if it works
Tho if it does work what do I do ? make branch and pr for archlinux ? Or make a branch and pr to edit the wiki theme ?

claymorwan commented:
Welp it does seems to be working, tho I'll have to do some fixes here and there. I also don't know if I should edit the wiki page's theme so it fits better with the aur theme, should I ?

claymorwan commented:
Oki I'm working on it, should I change the branch name too ?

uncenter commented:
No need to.

uncenter commented on scripts/userstyles.yml:
```diff
@@ -179,6 +182,14 @@ userstyles:
+    readme:
+      app-link: "https://archlinux.org/"
+    current-maintainers: [*claymorwan]
   arch-wiki:
```
You should also remove this entry.

uncenter commented on scripts/userstyles.yml:
```diff
@@ -179,16 +182,15 @@ userstyles:
-    past-maintainers: [*genshibe]
+      app-link: "https://archlinux.org/"
+    current-maintainers: [*claymorwan]
+  arch-wiki:
```
This needs to be removed.

uncenter commented on scripts/userstyles.yml:
```diff
@@ -179,16 +182,15 @@ userstyles:
-      app-link: "https://wiki.archlinux.org/"
-      usage: "Make sure to use the default **Light** theme"
-    current-maintainers: []
-    past-maintainers: [*genshibe]
```
You should move this past-maintainers line to the new combined arch-linux entry.

claymorwan commented:
So like, is it good now ? Or is there something else to change ?

uncenter commented:
At the core yes. I'm still seeing some unthemed or incorrectly themed things:

On a page like https://archlinux.org/download/, https://aur.archlinux.org/packages/follow-appimage, or just https://archlinux.org/packages/, the primary layer is the box in the center, which should have a background of base instead of surface1. Area around the box should be mantle.

The text of the current page/tab in the header is unthemed.
![CleanShot 2024-10-24 at 10 39 29](https://github.com/user-attachments/assets/0e38624d-06a5-4147-a0a1-0b6f392eb6e3)

On the home page, wiki, and security page: lots of unthemed dashed/solid horizontal lines/borders.

![CleanShot 2024-10-24 at 10 40 29](https://github.com/user-attachments/assets/8f47046c-0973-4ce8-8b96-ad20f7ffdccf)
![CleanShot 2024-10-24 at 10 41 17](https://github.com/user-attachments/assets/ce79f50d-906c-41cf-9b05-f1eed79ba44f)
![CleanShot 2024-10-24 at 10 42 00](https://github.com/user-attachments/assets/ef34834b-8e99-4151-beea-a51c4c50ec58)

High vulnerability severity text on the security page is unthemed (should be peach, class is .orange):

![CleanShot 2024-10-24 at 10 42 18](https://github.com/user-attachments/assets/277042ac-5262-4e73-a6c3-ca5452ada739)

claymorwan commented:
>On a page like https://archlinux.org/download/, https://aur.archlinux.org/packages/follow-appimage, or just https://archlinux.org/packages/, the primary layer is the box in the center, which should have a background of base instead of surface1. Area around the box should be mantle.

So you mean change every .box to have base as background, and every background of the page as mantle ?
Like that ? 
![image](https://github.com/user-attachments/assets/4352d1c5-df47-44bc-9a80-40b7d45e39ef)

uncenter commented:
> So you mean change every .box to have base as background, and every background of the page as mantle ?

Yep just like that.

> Been trying to theme these but I have no idea how to, tried border-color but nah

`border-bottom-color` of: `#news h4` for the dotted ones. `.mw-heading1, h1, .mw-heading2, h2` for the solid wiki ones.

claymorwan commented:
Chat is it gud

GitMuslim commented:
I'm not sure why you've requested a review from me, but either way looks good.

Though I thought we already had an aur theme, or was that for the wiki?

uncenter commented:
> I'm not sure why you've requested a review from me, but either way looks good.
> 
> 
> 
> Though I thought we already had an aur theme, or was that for the wiki?

They accidentally deleted the folder for the Stylus userstyle, triggering the CI to request a review from you since you maintain that.

GitMuslim commented:
> I suggested combining the proposed Arch User Repository userstyle with the existing Arch Wiki userstyle to create a unified Arch Linux userstyle, and that is what we have now here.

Sounds good.

claymorwan commented:
So do I have to do stuff again for the pr or nah ?

uncenter commented:
> So do I have to do stuff again for the pr or nah ?

Nothing right now, just needs reviewing.

claymorwan commented:
> > So do I have to do stuff again for the pr or nah ?
> 
> Nothing right now, just needs reviewing.

Oki, I'll be waiting then

claymorwan commented:
O wairt bu reviewing does that mean Inhabe to do the "request review" thingy

uncenter commented:
Nope, nothing you need to do right now.

isabelroses reviewed (changes requested):
The text colour here should be subtext0
![{D996B83F-1B89-419D-A3DB-374859302DEF}](https://github.com/user-attachments/assets/fb865797-b2e5-4aff-9a88-8beb0d878dae)
reproduction link: https://archlinux.org/news/mkinitcpio-hook-migration-and-early-microcode/

This should be text, its on the main page
![{6C306493-EBA5-4B68-85F6-781235E1ABA0}](https://github.com/user-attachments/assets/7aa04e53-c0a6-48a5-b6d6-a340574de13d)

The green here is wrong
![{D66140C1-CBA0-4E62-9BC6-81C552617F45}](https://github.com/user-attachments/assets/bd150564-9a70-4400-afdf-50d2963c400d)
repro link: https://archlinux.org/packages/extra/x86_64/syslog-ng/

Link hover shouldn't be so grey
![{6C279845-0ED4-4DF7-B0FD-84901E6BF916}](https://github.com/user-attachments/assets/eeea890e-8216-4d3c-a8d7-2674ff858f78)
repro url: https://bbs.archlinux.org/viewforum.php?id=23

the wiki seems to have some issues with the icons 
![{EC6D616A-1032-4579-9476-90686C25F19E}](https://github.com/user-attachments/assets/a22d38d2-7ef4-46a4-903c-b66f60f68e6e)

claymorwan commented:
> The text colour here should be subtext0

Fixed that.

> This should be text, its on the main page

I can't change that one, for some reason only 2 pages has the `.anb-selected` class name, so until they add it to all pages I can't do something about it (unless there's another that idk). In the meantime I changed them all to subtext0 and the selected one will have to be page's white text color.

> The green here is wrong

Couldn't find the class name or id for it cuz idk where to find a recently updated package (this green text doesn't seems to be on the aur).

> Link hover shouldn't be so grey

Istg I tried everything, but I can't change that hover color, I an changer the background color on hover but not the text color.

> the wiki seems to have some issues with the icons

These are on the wiki page and were already broken with the old arch wiki theme. I tried to fixed them but couldn't.

claymorwan commented:
Why tf are my commits unverified my email in the gpg key correct wth why

claymorwan commented:
O sorry, I thought I was supposed it keep it updated

isabelroses reviewed (changes requested):
On archwiki this seems broken.
![image](https://github.com/user-attachments/assets/0dea7129-aafc-4770-923e-9415167e6025)

Also the current archwiki themes the active navbar page with the accent colour.
![image](https://github.com/user-attachments/assets/e9c05974-a232-4ad6-a57e-8efc338e2de2)

Hovering links also looks somewhat poor, perhaps we could darken the accent colour?

claymorwan commented:
> On archwiki this seems broken. ![image](https://private-user-images.githubusercontent.com/71222764/388283264-0dea7129-aafc-4770-923e-9415167e6025.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIxNzYwMzgsIm5iZiI6MTczMjE3NTczOCwicGF0aCI6Ii83MTIyMjc2NC8zODgyODMyNjQtMGRlYTcxMjktYWFmYy00NzcwLTkyM2UtOTQxNTE2N2U2MDI1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIxVDA3NTUzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY3MDdkNTJmYTk3OWIxY2Q4YjFmN2RlY2U3Y2M2ZmRkNDMyMGQxN2ZlNzkwNGI2MTJkZWYxZmVhZGRmZGRhMGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ZJABVd6NYKxI8C6Hlf7rMnudehuamEjGzP0alA7x4do)

If u're talking about the icon then I can't change that, it was like that with the old arch wiki theme and I haven't found a way to change it

> Also the current archwiki themes the active navbar page with the accent colour. ![image](https://private-user-images.githubusercontent.com/71222764/388283403-e9c05974-a232-4ad6-a57e-8efc338e2de2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIxNzYwMzgsIm5iZiI6MTczMjE3NTczOCwicGF0aCI6Ii83MTIyMjc2NC8zODgyODM0MDMtZTljMDU5NzQtYTIzMi00YWQ2LWE1N2UtOGVmYzMzOGUyZGUyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIxVDA3NTUzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNiZDZkMzhlOTk0MTMzM2IxYmZkYTllNjU4NWY2ZmI0M2M1MzNmYWU0YjdjOTIxMzNiYTk1ZTM4MDZmZWUyMjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.AQ83YIhCOMUDKNRGdN84tqtNej48Hi8NGK8tdVYrtXA)

If you're talking about changing the white to accent color, I can only do that to like 2-3 of the pages, cuz these have a class name for when they're selected, the other pages doesn't tho

> Hovering links also looks somewhat poor, perhaps we could darken the accent colour?
So like change the hover color to every link ?

claymorwan commented:
> Homepage RSS feed buttons and current tab ![image](https://private-user-images.githubusercontent.com/52550063/396614573-3765c57d-e720-437d-a825-c9bd0f9bd8ef.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ2MTg1ODEsIm5iZiI6MTczNDYxODI4MSwicGF0aCI6Ii81MjU1MDA2My8zOTY2MTQ1NzMtMzc2NWM1N2QtZTcyMC00MzdkLWE4MjUtYzliZDBmOWJkOGVmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE5VDE0MjQ0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBjZjU1ZTg5NGZmNTZjZmIyMWJhMTMwMTNmNDVkOTRkMzBmNDEyZTA5NTFlNmYyNDhhYmVmYTRlZWRjYWY3MGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Th1RZ0sRVjppN6662EdXc2_7E62WZFqVqLywQ0__yxc) ![image](https://private-user-images.githubusercontent.com/52550063/396614648-5b7905f7-d501-4efe-a8a0-c637a993ea1a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ2MTg1ODEsIm5iZiI6MTczNDYxODI4MSwicGF0aCI6Ii81MjU1MDA2My8zOTY2MTQ2NDgtNWI3OTA1ZjctZDUwMS00ZWZlLWE4YTAtYzYzN2E5OTNlYTFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE5VDE0MjQ0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRiZTg3NzQyYTdmMDU4ZDI1OTliYmZkZjAwNjNiNDdhOTkyNTY3NGI2NTgyYzk4MDQ5YjY4OTgyZDdlODE3Y2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.q8V9sKMy-Ktfs_Ydlaut2WE1C79A_YCkubyQOSP65pM)
I tried to change that but it doesn't wanna idk what to do
> Visited pages highlight, language picker on the wiki & unthemed buttons for the dark mode selector and sidebar dropdown ![image](https://private-user-images.githubusercontent.com/52550063/396615034-56ff415d-d48c-4cd7-a777-5e1381f768ff.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ2MTg1ODEsIm5iZiI6MTczNDYxODI4MSwicGF0aCI6Ii81MjU1MDA2My8zOTY2MTUwMzQtNTZmZjQxNWQtZDQ4Yy00Y2Q3LWE3NzctNWUxMzgxZjc2OGZmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE5VDE0MjQ0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBjMDNmMjRkM2FjMjFjMjgxZjBjY2RhM2Y0ZTQ1MzkyZDU3NGY4NmE3M2UwZmNkMjVhMTEwODY1NGNiZTE4ZmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.LjWBjXouIhJ0kEABlcDZ31WBJTs-YGMfrSusIeHfiYs)
I should be able to change the highlights, language picker, tho same problem for the button as the RSS idk how to do it
> Speaking of the dark mode selector. The wiki is the only page that has one so it would make sense to respect it if it's set to dark otherwise use the user's set theme?
How do I even do that

I'll try to get to it next week, I'm still in my exams week but in vacation next week

uncenter commented on styles/arch-linux/catppuccin.user.less:
```diff
@@ -0,0 +1,1228 @@
+      box-shadow: @blue !important;
+    }
+
+    .archwiki-template-box-tip {
+      background-color: fadeout(hsla(90, 90%, 65%, 0.3), 10%) !important;
+      border-color: fadeout(hsla(90, 90%, 75%, 0.3), 10%);
+      color: @subtext1;
+    }
+
+    .archwiki-template-box-note {
+      background-color: fadeout(hsla(208, 100%, 61%, 0.3), 5%) !important;
+      border-color: fadeout(hsla(208, 100%, 81%, 0.3), 5%) !important;
+      color: @subtext1;
+    }
+
+    .archwiki-template-box-warning {
+      background-color: fadeout(hsla(7, 99%, 60%, 0.3), 5%) !important;
+      border-color: fadeout(hsla(7, 99%, 80%, 0.3), 05%) !important;
+      color: @subtext1;
+    }
```
The linter is going to fail because of a) `fadeout` and b) `hsla`. Instead, something like `fade(@green, 30%)` should be used (for the first one at least, and then make the colors match for the other two. Make sure to check how this looks on a page with some of these boxes.

uncenter commented:
Visited links shouldn't be gray (lavender I think):

![CleanShot 2025-02-02 at 12 39 18](https://github.com/user-attachments/assets/9a59fbb8-1705-4280-90a5-6f5ca1aca747)

The text of the current page in the header is an unthemed white:

![CleanShot 2025-02-02 at 12 39 54](https://github.com/user-attachments/assets/9d3d1967-9625-4593-a8d3-bb073ffb00b2)

![CleanShot 2025-02-02 at 12 39 59](https://github.com/user-attachments/assets/4260b1b1-dc1f-43a8-b3fd-40a5dbf4be1a)

The text of other not-current pages in the header on hover is also unthemed white:

![CleanShot 2025-02-02 at 12 40 06](https://github.com/user-attachments/assets/eed32ea0-438b-4d5c-bfeb-95cb260cc51c)

The lines in the boxes in red are unthemed (wiki main page), and the lines in the orange boxes are themed, but should be something like `overlay1` instead of `accent`.

![CleanShot 2025-02-02 at 12 40 56](https://github.com/user-attachments/assets/45e8b5d4-1f8a-4311-bf73-265476058472)

uncenter commented:
Links on hover and active hover are unthemed, as well as this bar for the table of contents.

![CleanShot 2025-02-02 at 12 43 11](https://github.com/user-attachments/assets/aad0018e-53f3-45d6-b7b1-c21067bd89bf)

justbispo commented:
Text in table cells are impossible to read
![archwiki](https://github.com/user-attachments/assets/dfd7891a-b1c6-450c-a78c-9b2ea21d8a47)

uncenter commented:
https://gitlab.archlinux.org/archlinux/archwiki/-/merge_requests/89 was merged, meaning the wiki now uses the latest Mediawiki version - which ultimately means it now uses the newly introduced CSS variables that https://github.com/catppuccin/userstyles/pull/1075 uses on Wikipedia. We should be able to reuse our Wikipedia port (might be worth splitting it out into its own repository catppuccin/mediawiki and importing the CSS in each userstyle that makes use of it?) for it now, but one remaining issue is that Arch Linux wiki has their own custom CSS that still overrides some variables. If someone wants to update the custom CSS for https://gitlab.archlinux.org/archlinux/archwiki to set the new CSS variables, it will then be compatible with our userstyle and make future maintenance a breeze!

Thomas-Philippot commented:
>  If someone wants to update the custom CSS for https://gitlab.archlinux.org/archlinux/archwiki to set the new CSS variables, it will then be compatible with our userstyle and make future maintenance a breeze!

I can make a branch for this, new variables are `skin-theme-clientpref-{os/day/night}` ?

uncenter commented:
> > If someone wants to update the custom CSS for [gitlab.archlinux.org/archlinux/archwiki](https://gitlab.archlinux.org/archlinux/archwiki) to set the new CSS variables, it will then be compatible with our userstyle and make future maintenance a breeze!
> 
> I can make a branch for this, new variables are `skin-theme-clientpref-{os/day/night}` ?

I'm not sure how their archwiki repository is set up, but if there is a way for their custom CSS use the [standard Wikipedia CSS variables](https://github.com/catppuccin/userstyles/blob/a751e469ee2565de14e69d91568b1efe1629d091/styles/wikipedia/catppuccin.user.less#L179-L256) so that our CSS can properly override their CSS, that would make this work.

claymorwan commented:
> We should probably close this one now the new PR has been opened :]

yea i highkey completely forgot abt this, pretty sure the new PR gonna do a better job than me at this too
